dlrm_s_pytorch.py

```diff
@@ -566,6 +566,8 @@ def collate_wrapper(list_of_tuples):
                 return list_of_tuples[0]
 
         ln_emb = np.fromstring(args.arch_embedding_size, dtype=int, sep="-")
+        if args.emulate_8_gpu:
+            ln_emb = ln_emb[:int(ngpus * ln_emb.size/8)]
```

Comment:
Does it not split the tables even in single gpu case? For example, if I want 16 tables in a 1-gpu case, where `ln-emb=1000-1000....1000`, it creates only 2 tables on the GPU?

Reply:
By "it", do you mean the "emulate_8_gpu" case, or the original script? If it's the former, then yes, that's exactly what's happening here. If it's the latter, then no, because if you were to run the script on a single-gpu, for eg. by setting HIP_VISIBLE_DEVICES=0, then it'd try to put all the 16 embedding tables on that GPU.

Reply:
I meant `emulate_8_gpu` case. With your elucidation below, wherein the input config script will have params of  8 gpu case, this makes sense to me. `ln_emb` in the current config scripts we have needs to be changed I guess?

Reply:
Yes, `ln_emb` in the config script would need to be for an 8GPU run